spec/spec_helper.rb

```diff
@@ -17,4 +19,7 @@
 
 RSpec.configure do |config|
   config.include StubHelpers
+  config.before(:each) do
+    GovukDependencies.cache.clear
```

Comment:
Just so I'm clear, how does this differ from `GovukDependencies.cache.flush_all`?

Reply:
[`.flush_all`](https://www.rubydoc.info/gems/dalli/2.5.0/Dalli%2FClient:flush) is a method on dalli/memchached.
[`.clear`](https://api.rubyonrails.org/classes/ActiveSupport/Cache/Store.html#method-i-clear) is a method on acitvesupport/filestore.

Reply:
Thanks. It's a bit of a smell to have to call different methods on `GovukDependencies.cache` depending on which underlying store is currently in use - means the developer has to keep track of that context, and there's an increased risk of introducing bugs by calling a particular method that isn't available in certain contexts.

I'd suggest using the [delegator pattern](https://ruby-doc.org/stdlib-2.5.1/libdoc/delegate/rdoc/SimpleDelegator.html) to encapsulate that logic in the CacheStore module, e.g.

```ruby
class CacheStore::Store < SimpleDelegator
  def initialize(cache)
    super
    @cache = cache
  end

  def clear
    @cache.respond_to?(:flush_all) ? @cache.flush_all : @cache.clear
  end
end
```

That would mean we only ever call `GovukDependencies.cache.clear`, and let the store itself decide which underlying method to call. The 'delegator' bit of it means any methods not defined in our `CacheStore::Store` class are delegated to the underlying cache lib (e.g. `GovukDependencies.cache.foo` would call `@cache.foo`).

Reply:
Very nice, thanks Chris I'll make that change.